### ISSUE TITLE
[ReactorCraft] [v26b] Pebble bed reactors (HTGR) shape cache not updating properly?

### ISSUE THREAD

**Issue by Vectrobe** [ReikaKalseki/Reika_Mods_Issues#2559]
The v26b patch appears to have created a regression where the reactors end up gradually ticking heat up, followed by a huge drop in heat across the reactor. The highest possible temperature to attain appears to only be 350C with about 32 cores in a 9x9 fashion, CO2 exchangers in the centre.

It also appears that building more cores causes completely disconnected existing cores to go stone-cold, until the next server reboot or save re-load that resets this behaviour and restores the existing temperatures. IE you have a constant counter in use instead of scanning attached blocks?

**Comment by ReikaKalseki** [comment]
I greatly increased atmospheric heat loss in v25 or v26. However, insulated reactions should behave similarly to before.

And no, there is no constant counter; they do work off of attached blocks.

**Comment by Vectrobe** [comment]
What insulation block did you test with? I tested one wrapped in the standard reactorcraft concrete and it only got to about 80C with 16 cores...

**Comment by ReikaKalseki** [comment]
There is no tiering of thermal insulation; as long as it is a proper full block, it insulates. I usually just use some glass variant.

**Comment by Vectrobe** [comment]
Well, another round of testing and I ended up getting a 32 core, insulated reactor to work at 750-800C. However to do so involved removing all existing reactors > exiting the world and re-entering > rebuild the whole reactor, being sure that each core is "facing" the centre heat exchanger stack (8 exchangers, single tower) > exiting world and re-starting full client.
After restarting the client again the reactors work, but so long as you don't change any blocks or build another one elsewhere. So that block tracking issue needs fixing for sure.

Another note though, this 32 core tower seems to produce about 5-6 full turbines worth of steam, but just under the requirement for a high pressure turbine. That seems like way too much heat production? the general thing with these reactors is they produce less total heat, which is what makes them safer as the risk of meltdown is very low, which in turn means that most IRL implementations consist of multiple towers for only 10-200MW.

Assuming the [powered] heat exchanger uses real heat conversion numbers, I think the C02 heat exchangers could do with a little nerf in throughput, and possibly the core heat little buff in heating. The general expectation is that the cores should be able to heat up fast, but gas cooling is what limits the design throughput, as any given inert gas has much lower capacity and conductivity than water. I think the heat of the output C02 should only be around 200-300C if the cores were 800C.

**Comment by ReikaKalseki** [comment]
My typical HTGR, with 24 cores, is only just capable of managing two LP turbines.

**Comment by MalTeeez** [comment]
Mine which has 24 cores can produce enough steam for about 3+3/4 Turbines

**Comment by 3bygone3** [comment]
Upon finding out that I couldn't get my HTGR working in v26b I tested couple previous versions. V26b, v26a and v25 all behaved in similar manner, where 3x3x5 and 5x5 reactors with exchangers in the middle and insulated with stone never reached 800C no matter how I built them, or how many times I restarted the game. V24 worked fine with both versions. Tested with just dragonapi, rotarycraft, reactorcraft, enderIO, enderCore (for conduits) and waila (so I can see the temperature easier).

**Comment by XCOAnnika** [comment]
Still in 27

**Comment by ReikaKalseki** [comment]
I looked at the logic again - it is unchanged since long before v25 and is using the normal - and again unchanged since *2015* - block tracking.

**Comment by 3bygone3** [comment]
Well, it still doesn't seem to work properly in v29c. Only [this](https://github.com/ReikaKalseki/Reika_Mods_Issues/issues/2559#issuecomment-640214828) results in working reactor for me past v24a and working reactors from this old version quickly cool down after updating to newer versions.

**Comment by ReikaKalseki** [comment]
Are they improperly insulated?

Looking at it, all of the common conduit and piping blocks count as air now - that likely means that previously "insulated" reactors no longer count as such.

**Comment by 3bygone3** [comment]
That depends on whether only sides needs to be insulated or also top and bottom, sides are insulated with one layer of standard stone (including corners), top and bottom is covered only by conduits.

**Comment by ReikaKalseki** [comment]
All sides need to be insulated.

**Comment by 3bygone3** [comment]
This is what I am testing on:
![image](https://user-images.githubusercontent.com/10800723/112715670-eecb0380-8ee1-11eb-95e0-89953916c5ac.png)
![image](https://user-images.githubusercontent.com/10800723/112715676-01ddd380-8ee2-11eb-9680-dda77e717c3d.png)
Left it running for about half an hour, highest temp on the bottom/top cores I've seen was around 400C. Tried covering the conduits with facades but that didn't help. Downgraded to v24c and within couple seconds it reached working temperature.

**Comment by ReikaKalseki** [comment]
Yes, those very much count as noninsulated on the top and bottom surfaces.

Adding facades does not change the conduit block type; they remain a "conduit" block.

**Comment by 3bygone3** [comment]
That is good to know. After some testing it seems that AE's export/import buses works, however I am unsure if thats intended. Also there clearly are some issues with reactor structure (?) updates while doing modifications, got to a point where heat exchangers refused to heat up at all from surrounding cores until I broke and placed them down again. All of this leaves me wondering what is the intended way to fuel and extract spent fuel from these reactors. Penumatic item pumps seemed to insulate fine, but their requirement to be powered makes the reactor design somewhat limited.
